dotCMS/src/main/java/com/dotmarketing/business/ajax/RoleAjax.java

```diff
@@ -128,7 +132,7 @@ protected RoleAjax(SystemEventsAPI systemEventsAPI) {
 
 	}
 
-	private Map<String, Object> constructRoleMap(Role role, String[] excludeRoles, boolean onlyUserAssignableRoles) throws DotDataException {
+	private Map<String, Object> constructRoleMap(Role role, List<String> excludeRoles, boolean onlyUserAssignableRoles) throws DotDataException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)